File: BlogServer/Blog.Model/Param/ArticleListParam.cs


namespace Blog.Model.Param
{
    public class ArticleFindParam: PageParam
    {
        //[DefaultValue(null)]
        public int? ClassId { get; set; }
        //[DefaultValue(null)]
        public string? Title { get; set; }
    }


    public class ArticleInsertParam
    {
        public required string[] Cover { get; set; }
        public required string Title { get; set; }
        public required string Content { get; set; }
        public required int ClassId { get; set; }
        public required int CreateUserId { get; set; }
    }

    public class ArticleUpdateParam: IDParam
    {
        public required string[] Cover { get; set; }
        public required string Title { get; set; }
        public required string Content { get; set; }
        public required int ClassId { get; set; }
    }


    public class ArticleFindHomeParam : PageParam
    { 
        public int? ClassId { get; set; }
        public string? Search { get; set; }
    }


    public class ArticleAndRandomFindHomeParam : PageParam
    {
        public string? Search { get; set; }
    }

}

File: blog-server/src/module/admin/articleList/model/ArticleList.param.ts
import {IsString, IsInt, IsArray, IsOptional} from 'class-validator'
import {IdParam, PageParam} from "../../../../model/BaseParam";
// import {ApiObjectProperties} from "../../../../common/ApiObjectProperties";

// @ApiObjectProperties
export class InsertArticleParam {
    @IsArray({ message: 'cover必须是数组类型' })
    cover: string[];

    @IsString({ message: 'title必须是字符类型' })
    title: string

    @IsString({ message: 'content必须是字符类型' })
    content: string

    @IsInt({ message: 'classId必须是字符类型' })
    classId: number;

    @IsInt()
    createUserId: number
}


export class ArticleFindParam extends PageParam{
    @IsInt()
    @IsOptional()
    classId?: number

    @IsString()
    @IsOptional()       // 表示 search 字段是可选的
    title?: string
}


export class UpdateArticleParam extends IdParam{

    @IsArray({ message: 'cover必须是数组类型' })
    cover: string[];

    @IsString({ message: 'title必须是字符类型' })
    title: string

    @IsString({ message: 'content必须是字符类型' })
    content: string

    @IsInt({ message: 'classId必须是字符类型' })
    classId: number;
}
